src/index.tsx

```diff
@@ -13,26 +17,30 @@ const container = document.getElementById('root')
 const sw = await navigator.serviceWorker.register(new URL('sw.ts', import.meta.url))
 const root = ReactDOMClient.createRoot(container)
 
-// get urlSearchParameters and check for existing helia-sw parameter & value
-const urlSearchParams = new URLSearchParams(window.location.search)
-const heliaSwPath = urlSearchParams.get('helia-sw')
-if (heliaSwPath != null) {
+const subdomain = window.location.hostname.replace(`.${BASE_URL}`, '')
+const subdomainRegex = /^(?<origin>[^/]+)\.(?<protocol>ip[fn]s)?$/
+const subdomainMatch = subdomain.match(subdomainRegex)
+
+const isPathBasedRequest = window.location.pathname.startsWith('/ip')
+const isSubdomainRequest = subdomainMatch?.groups != null
+
+if ((isPathBasedRequest || isSubdomainRequest)) {
```

Comment:
How does this check for whether the SW has registered/triggered work?  Wouldn't `isPathBasedRequest || isSubdomainReques` be the same?

Reply:
I added some tests & comments which show more about what is being checked.

If the service worker is registered, and it's a valid path, the request is intercepted by the service worker.

If the service worker is not registered, it will reload the page and the service worker will intercept things after the reload.

There could still be some more intelligent determinism for the sw registration & redirect logic.